lib/Optimizer/Transforms/PySynthCallableBlockArgs.cpp

```diff
@@ -59,6 +59,41 @@ class ReplaceCallIndirect : public OpConversionPattern<func::CallIndirectOp> {
   }
 };
 
+class ReplaceCallCallable
+    : public OpConversionPattern<cudaq::cc::CallCallableOp> {
+public:
+  const std::vector<std::string> &names;
+  const std::map<std::size_t, std::size_t> &blockArgToNameMap;
+
+  ReplaceCallCallable(MLIRContext *ctx,
+                      const std::vector<std::string> &functionNames,
+                      const std::map<std::size_t, std::size_t> &map)
+      : OpConversionPattern<cudaq::cc::CallCallableOp>(ctx),
+        names(functionNames), blockArgToNameMap(map) {}
+
+  LogicalResult
+  matchAndRewrite(cudaq::cc::CallCallableOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto callableOperand = adaptor.getCallee();
+    auto module = op->getParentOp()->getParentOfType<ModuleOp>();
+    if (auto blockArg = dyn_cast<BlockArgument>(callableOperand)) {
+      auto argIdx = blockArg.getArgNumber();
+      auto replacementName = names[blockArgToNameMap.at(argIdx)];
+      auto replacement = module.lookupSymbol<func::FuncOp>(
+          "__nvqpp__mlirgen__" + replacementName);
+      if (!replacement) {
+        op.emitError("Invalid replacement function " + replacementName);
```

Comment:
Should the pass itself fail here? Or should it just not do the replacement? In other words, should this pass just be best effort (like most passes) or should it be an ultimatum: it shall be replaced?

Reply:
hadn't thought of that originally, updated to just not perform the replacement, no error thrown